### DISCUSSION TITLE
Common java main method

### DISCUSSION THREAD

**Discussion by BritneyStars** [apache/skywalking#8302] [Q&A] [answered]
In project, it is a common project ,and not use some framework,such as spring. I define a method  add @Trace annotation and run in main method called. But I found in segment indice can find the call record ,but in metric indice,the record can not find. I want to kown how to calculate common method metrics

**Comment by wu-sheng** [comment, marked as answer]
What do you want to do? There is no guarantee `main`'s trace could be collected in time. I think one side you just got luck, the other side, what metric do you want? It is just a method, nothing special to analysis.

Notice, **what is SkyWalking**

```
SkyWalking: an APM(application performance monitor) system, especially designed for microservices, cloud native and container-based architectures.
```

**Comment by BritneyStars** [comment]
There is a batch task, run in main method and configurate the crontab time in linux . Now want to record the batch task execute time in process and some sqls and component methods. In batch task, may use mysql, elasticsearch and other component, so hope analysis the duration metric

**Reply by wu-sheng** [reply]
Check the logic endpoint doc, I think you would like that concept.

**Comment by BritneyStars** [comment]
Thank you ,I got it .
